tests/integration/targets/nd_backup_schedule/tasks/main.yml

```diff
@@ -0,0 +1,282 @@
+      - nt_add_backup_schedule_5 is failed
+      - nt_add_backup_schedule_5.msg == "ND Error 400{{':'}} field 'encryptionKey' is invalid. It must contain at least one letter and one number, and have a minimum length of 8 characters"
+
+# QUERY
+- name: Query a backup schedule 1
+  cisco.nd.nd_backup_schedule:
+    output_level: debug
+    name: backupschedule1
+    state: query
+  register: query_one
+
+- name: Query all backup schedules
+  cisco.nd.nd_backup_schedule:
+    output_level: debug
+    state: query
+  register: query_all_1
+
+- name: Assertion check for query a backup schedule
+  ansible.builtin.assert:
+    that:
+      - query_one is not changed
+      - query_one.current.frequency == 7
+      - query_one.current.name == "backupschedule1"
+      - query_one.current.remoteLocation == "test"
+      - query_one.current.startTime == "2025-01-02T15:04:05Z"
+      - query_one.current.type == "configOnly"
+      - query_one.current.user == "admin"
+      - query_all_1 is not changed
+      - query_all_1.current.0.frequency == 7
+      - query_all_1.current.0.name == "backupschedule1"
+      - query_all_1.current.0.remoteLocation == "test"
+      - query_all_1.current.0.startTime == "2025-01-02T15:04:05Z"
+      - query_all_1.current.0.type == "configOnly"
+      - query_all_1.current.0.user == "admin"
+      - query_all_1.current.1.frequency == 30
+      - query_all_1.current.1.name == "backupschedule2"
+      - query_all_1.current.1.remoteLocation == "test"
+      - query_all_1.current.1.startTime == "2025-01-03T19:04:05Z"
+      - query_all_1.current.1.type == "configOnly"
+      - query_all_1.current.1.user == "admin"
+
+# DELETE
+- name: Delete a backup schedule 1 (check_mode)
+  cisco.nd.nd_backup_schedule: &cm_rm_backup_schedule
+    output_level: debug
+    name: backupschedule1
+    state: absent
+  check_mode: true
+  register: cm_rm_backup_schedule
+
+- name: Delete a backup schedule 1
+  cisco.nd.nd_backup_schedule:
+    <<: *cm_rm_backup_schedule
+  register: rm_backup_schedule
+
+- name: Delete a backup schedule 1 again
+  cisco.nd.nd_backup_schedule:
+    <<: *cm_rm_backup_schedule
+  register: rm_backup_schedule_again
+
+- name: Delete a backup schedule 2
+  cisco.nd.nd_backup_schedule:
+    <<: *cm_rm_backup_schedule
+    name: backupschedule2
+
+- name: Assertion check for delete a backup schedule
+  ansible.builtin.assert:
+    that:
+      - cm_rm_backup_schedule is changed
+      - cm_rm_backup_schedule.current == {}
+      - cm_rm_backup_schedule.previous.frequency == 7
+      - cm_rm_backup_schedule.previous.name == "backupschedule1"
+      - cm_rm_backup_schedule.previous.remoteLocation == "test"
+      - cm_rm_backup_schedule.previous.startTime == "2025-01-02T15:04:05Z"
+      - cm_rm_backup_schedule.previous.type == "configOnly"
```

Comment:
can we not add a test case with backup_type parameter?

Reply:
Done!